server/index.js

```diff
@@ -109,6 +113,54 @@ const getSalt = async event => {
   return { hashes: results }
 }
 
+/**
+ * Allows an authority to report an analysis
+ * @return {Promise<string>} Id of the report
+ */
+const reportAnalysis = async event => {
+  let authorization = event.headers['authorization']
+  if (!authorization) throw new ServerError("Missing header 'authorization'", 400)
+
+  let { date } = JSON.parse(event.body)
+  date = moment(date)
```

Comment:
Is adding `moment` a bit heavy weight for validating a date?

What about `isNaN(new Date(date)))`?

Reply:
You are right, I added to manage dates easier, but in the end I only used it for that